
<h3>"Run ShortBRED" tab:  EFI-CGFP Input SSNs</h3>

<div class="tutorial_figure">
<img src="tutorial/input_tab.png" width="100%" alt="Image of the input form." />
</div>

<p>
In the "Run ShortBRED" tab, the user uploads the colored SSN xgmml file for 
which the markers are to be identified (red arrow; either compressed/zipped or 
uncompressed). The SSN should have been segregated into isofunctional clusters 
so that EFI-CGFP will allow orthologues of the members of these clusters to be 
identified in the metagenome datasets.
</p>

<p style="color: red">
<b>
The input SSN MUST be a Colored SSN generated with either the Color SSN utility 
of EFI-EST or EFI-GNT.  The node attributes for a Colored SSN include "Cluster 
Number" and "Singleton Number" for each (meta)node [the former if the 
(meta)node is located in a cluster with ≥2 accession IDs; the latter if the 
(meta)node is a singleton (a single accession ID)].  These numbers are used to 
identify the clusters and singletons with metagenome hits both in SSN, heat 
maps, and tables generated by the EFI-CGFP quantify step ("Quantify Results" 
page, vide infra).
</b>
</p>

<p>
Before they are colored, the SSNs can be of two types:  1) SSNs generated with 
Options A, B, or D of EFI-EST with sequences derived entirely from the UniProt 
database
(<a href="https://www.uniprot.org/">https://www.uniprot.org/</a>)
and 2) SSNs generated with Option C of 
EFI-EST that include sequences provided in a user-provided FASTA file so that 
they need not be included in the UniProt database, e.g., in the NCBI database 
or proprietary sequences obtained from in-house sequencing projects. In 
generating the original SSN, the user may elect to read the FASTA headers to 
obtain node attributes if the accession ID can be found in the UniProt 
database; however, these node attributes are not used by EFI-CGFP.
</p>

<p>
For the latter type of SSNs, the sequences provided in the FASTA file are used 
to generate the CD-HIT 85 clusters in the EFI-CGFP pipeline; the SSNs also may 
include nodes/sequences for a user-specified UniProt or InterPro family (using 
the Advanced Options of Option C).  EFI-CGFP only uses the sequences to 
generate the CD-HIT 85 clusters from which the markers are identified and used 
to interrogate the metagenome datasets in the Quantify step; it does not use 
any of the node attributes.
</p>

<p>
Either type of SSN can be a full SSN with a node for each sequence/accession ID 
or a representative node (rep node) SSN in which the sequences/accession IDs 
are grouped in metanodes that share specified levels of sequence identity by 
EFI-EST. Depending on the RAM available on the user’s computer, the user may 
not be able to view a full SSN, but the user will be able to view a rep node 
SSN. We recommend the highest resolution rep node SSN that can be manipulated 
with Cytoscape on the user’s computer (largest possible sequence identity) so 
that proteins with different functions can be expected to be located in 
separate SSN clusters. EFI-CGFP "expands" the metanodes in rep node SSNs to
include all of the sequences in the identification of markers
</p>

<p>
We also recommend that users apply a minimum length filter to ensure that the 
sequences are "full length" when the input SSN is generated with EFI-EST.  An 
appropriate minimum length can be selected by examining the length histogram 
provided by EFI-EST.  Rep node SSNs are generated after the minimum length 
filter is applied; therefore, all of the sequences in rep node SSNs will be 
"full length". The 
marker identification step in ShortBRED involves initial clustering of proteins 
into groups of sequences that share &ge;85% sequence identity (CD-HIT 85 
clusters); these sequences then are aligned (multiple sequence alignment using 
MUSCLE) to generate a consensus sequence.  Finally, the consensus sequence is 
used to identify unique markers for each CD-HIT 85 cluster.  The presence of 
sequence "fragments" may bias the multiple sequence alignment/identification of 
the consensus sequence so they should be avoided in/absent from the input SSN.   
EFI-CGFP maps the abundance of metagenome hits to the markers and then to the 
SSN clusters that contain the CD-HIT 85 sequences with the markers.
</p>

<p>
EFI-EST uses UniRef90 clusters when a user-specified family contains ≥25,000 
sequences (Options B, C, and D).  EFI-CGFP "expands" UniRef90 clusters to 
include all of the sequences in the identification of markers; however, the 
clusters can include fragments.  Because the minimum length filter in EFI-EST 
is applied only to the UniRef90 seed sequence, the same minimum length filter 
should be specified in the "minimum sequence length" window to remove 
fragments.  </p>

<p>
EFI-CGFP identifies and uses only unique sequences (100% sequence identity over 
100% of the length of each sequence) in the input SSN so that the consensus 
sequence is not biased by multiple occurrences of the same sequence; metanodes 
in rep node SSNs are expanded so that all sequences included in the metanodes 
are used in the identification of unique sequences.
</p>

<p>
After the colored input SSN is uploaded, marker identification is initiated 
(with the "Upload SSN" button"’; blue arrow).  Because marker identification 
involves large-scale BLAST analyses, it is time consuming, e.g., ~24 hrs for 
the GRE superfamily (InterPro family IPR004184 with a minimum length filter of 
500 residues; average length ~700 residues; ~9000 unique sequences).  Marker 
identification for SSNs with larger numbers of sequences and/or longer 
sequences takes longer.
</p>

<p>
An e-mail is sent to the user when the input SSN has been uploaded and the 
marker identification has been initiated.  The "Previous Jobs" tab will display 
the job as soon as it is received and its status will be indicated as 
"PENDING"; when marker identification has been initiated, the job status will 
be changed to "RUNNING".  </p>


